### PR TITLE
buffs the traitor toolbox turret to actually be worth 11tc

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

the toolbox turret now has an unique sprite by infraredbaron
it can now be only deployed with a combat wrench (included in the toolbox)
it fires way quicker, with a bit lower power, but faster shots (they dont hit you either)
it has more health and doesnt break from low integrity
it can be repaired with a combat wrench after a few seconds or folded back into toolbox form
adds said toolbox turret to the nukie uplink (nukie version ignores everything with syndicate faction)
https://streamable.com/0i3q07

## Why It's Good For The Game

this turret was always really laughable, it costed 11tc while being pretty much worthless
now it should be a bit fun, and for nukies its a nice area denial tool

## Changelog

:cl: Fikou, InfraredBaron
balance: the traitor toolbox turret is now worth something
expansion: nukies can now buy the toolbox turret
imageadd: new sprites for the traitor turret
/:cl:

### CODE REVIEW

**Comment by Amrabol** [comment]
Actual good traitor engi turret. Nice

**Comment by Farquaar** [comment]
Now all we need is the ability to use more gun by combining two turret toolboxes into one level 2 turret.

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -269,19 +269,93 @@ effective or pretty fucking useless.
...
 	new /obj/item/crowbar(src)
 	new /obj/item/analyzer(src)
 	new /obj/item/wirecutters(src)
 
-/obj/item/storage/toolbox/emergency/turret/attackby(obj/item/I, mob/living/user, params)
-	if(I.tool_behaviour == TOOL_WRENCH && user.combat_mode)
-		user.visible_message(span_danger("[user] bashes [src] with [I]!"), \
-			span_danger("You bash [src] with [I]!"), null, COMBAT_MESSAGE_RANGE)
+/obj/item/storage/toolbox/emergency/turret/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(istype(attacking_item, /obj/item/wrench/combat) && attacking_item.toolspeed && user.combat_mode && attacking_item.use_tool(src, user, 2 SECONDS, volume = 50))
```
Comment (Iamgoofball) [inline code/game/objects/items/devices/traitordevices.dm:279]: you should be allowed to do this with any wrench but slower if it's not a combat wrench

Comment (Fikou) [reply code/game/objects/items/devices/traitordevices.dm:279]: i dont want to let people facetank the bullets to steal the traitor item for themselves

Comment (Iamgoofball) [reply code/game/objects/items/devices/traitordevices.dm:279]: my concern is the traitor losing his special wrench and being unable to use his toolbox as a result

File: code/game/objects/items/devices/traitordevices.dm [hunk 2]
```diff
@@ -269,19 +269,98 @@ effective or pretty fucking useless.
...
+
+/obj/machinery/porta_turret/syndicate/toolbox/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == toolbox)
+		toolbox = null
 		qdel(src)
 
-	..()
+/obj/machinery/porta_turret/syndicate/toolbox/ui_status(mob/user)
+	if(faction_check(user.faction, faction)
```
Comment (Fikou) [inline code/game/objects/items/devices/traitordevices.dm:360]:
```suggestion
	if(faction_check(user.faction, faction))
```

**Comment by datorangebottle** [comment]
now what it needs is a USB port so I can make it automatically turn on/off if some other asshole enters my maint base.

**Comment by D4C-420** [comment]
There anything special about the combat wrench beyond it being the needed key to use the tool box?  Is there a reason for the wrench being able to turn on or off?

**Comment by Fikou** [comment]
> There anything special about the combat wrench beyond it being the needed key to use the tool box? Is there a reason for the wrench being able to turn on or off?

i didnt add it its from the nukie mech bundle

**Comment by ATH1909** [comment]
Can silicons remotely interface with/turn off a toolbox turret?

**Comment by Fikou** [comment]
> Can silicons remotely interface with/turn off a toolbox turret?

no

**Comment by B4CKU** [comment]
i love this so much

**Pull request merged**